nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/registry/StandardFlowRegistryClientNode.java

```diff
@@ -0,0 +1,353 @@
+import org.apache.nifi.registry.flow.FlowRegistryInvalidException;
+import org.apache.nifi.registry.flow.FlowRegistryUtil;
+import org.apache.nifi.registry.flow.RegisteredFlow;
+import org.apache.nifi.registry.flow.RegisteredFlowSnapshot;
+import org.apache.nifi.registry.flow.RegisteredFlowSnapshotMetadata;
+import org.apache.nifi.registry.flow.SimpleRegisteredFlowSnapshot;
+import org.apache.nifi.registry.flow.SimpleRegisteredFlowSnapshotMetadata;
+import org.apache.nifi.util.CharacterFilterUtils;
+import org.apache.nifi.util.file.classloader.ClassLoaderUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class StandardFlowRegistryClientNode extends AbstractComponentNode implements FlowRegistryClientNode {
+
+    private final FlowController flowController;
+    private final LoggableComponent<FlowRegistryClient> client;
+    private volatile String description;
+
+    public StandardFlowRegistryClientNode(
+            final FlowController flowController,
+            final LoggableComponent<FlowRegistryClient> client,
+            final String id,
+            final ValidationContextFactory validationContextFactory,
+            final ControllerServiceProvider serviceProvider,
+            final String componentType,
+            final String componentCanonicalClass,
+            final ComponentVariableRegistry variableRegistry,
+            final ReloadComponent reloadComponent,
+            final ExtensionManager extensionManager,
+            final ValidationTrigger validationTrigger,
+            final boolean isExtensionMissing) {
+        super(id, validationContextFactory, serviceProvider, componentType, componentCanonicalClass, variableRegistry, reloadComponent, extensionManager, validationTrigger, isExtensionMissing);
+        this.flowController = flowController;
+        this.client = client;
+    }
+
+    @Override
+    public Authorizable getParentAuthorizable() {
+        return flowController;
+    }
+
+    @Override
+    public Resource getResource() {
+        return ResourceFactory.getComponentResource(ResourceType.Controller, getIdentifier(), getName());
+    }
+
+    @Override
+    public String getProcessGroupIdentifier() {
+        return null;
+    }
+
+    @Override
+    protected List<ValidationResult> validateConfig() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void verifyModifiable() throws IllegalStateException {
+        // FlowRegistryClients has no running state
+    }
+
+    @Override
+    public void setProperties(final Map<String, String> properties, final boolean allowRemovalOfRequiredProperties, final Set<String> updatedSensitiveDynamicPropertyNames) {
+        super.setProperties(properties, allowRemovalOfRequiredProperties, updatedSensitiveDynamicPropertyNames);
+    }
```

Comment:
Any reason we're overriding this method just to call super?

Reply:
I think there was some -now obsolete- logic there. I will clean it up